### PR TITLE
added the ability to connect to ipv6 localhost

### PR DESCRIPTION
**Pull request by czardoz**

### CODE REVIEW

File: lib/requester/request-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if (!port) {
+            port = _.startsWith(request.url.protocol, 'https') ? '443' : '80';
+        }
+        port = _.parseInt(port, 10);
+
+        async.waterfall([
+            function (next) {
+                // Try checking if we can connect to IPv6 localhost ('::1')
+                connect(LOCAL_IPV6, port, function (err) {
+                    next(null, !err);
```
Comment (czardoz) [inline lib/requester/request-wrapper.js]: timeout

File: lib/requester/request-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    connect = function (host, port, callback) {
+        var socket = new Socket();
+
+        socket.once('connect', function () {
+            socket.destroy();
+            callback();
+        });
+
+        socket.once('error', callback);
+        socket.connect(port, host);
```
Comment (czardoz) [inline lib/requester/request-wrapper.js]: ensure that socket does not remain in memory once its done

File: lib/requester/request-wrapper.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    lookup = function (request, hostname, options, callback) {
+        if (hostname !== 'localhost') {
+            return dns.lookup(hostname, options, callback);
+        }
+        var port = _.get(request, 'url.port');
+
+        if (!port) {
+            port = _.startsWith(request.url.protocol, 'https') ? '443' : '80';
+        }
+        port = _.parseInt(port, 10);
```
Comment (czardoz) [inline lib/requester/request-wrapper.js]: protect batman

File: lib/requester/requester.js [hunk 4]
```diff
@@ -109,7 +109,7 @@ Requester.prototype.request = function (item, cb, scope) {
             };
 
         cb.call(scope || this, null, response, actualRequest, responseObj, request);
-    });
+    }, request);
```
Comment (czardoz) [inline lib/requester/requester.js]: brainwave - send in options

File: lib/requester/request-wrapper.js (hunk 2 above)
Comment (shamasis) [inline lib/requester/request-wrapper.js]: pending

**Pull request merged**